exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -219,12 +232,43 @@ public void sessionDestroyed(HttpSessionEvent se) {
           securityHandler.logout(sessionAuth);
           session.removeAttribute(SessionAuthentication.__J_AUTHENTICATED);
         }
+
+        // Clear all the custom attributes set as part of session
+        clearSessionCustomAttributes(session);
       }
     });
 
     return new SessionHandler(sessionManager);
   }
 
+  private void clearSessionCustomAttributes(HttpSession session) {
```

Comment:
WebUserConnection is a class which is encapsulating all the resources. I had the logic of close and cleanup in it earlier. But I Didn't set that as attribute of Http session object since in that case it will be reset for each request. Since there are not many resources which we are setting inside HttpSession and each of them are really a separate property for the session in itself, I guess keeping them separated should be fine.